com.unity.sg2/Editor/GraphUI/DataModel/ShaderGraphModel.cs

```diff
@@ -145,35 +146,18 @@ public void Init(GraphHandler graph, bool isSubGraph)
         {
             graphHandlerBox.Init(graph);
             this.isSubGraph = isSubGraph;
+            if (!isSubGraph)
+            {
+                Targets.Add(ShaderGraphAssetUtils.GetUniversalTarget());
+            }
 
             // Generate context nodes as needed.
             // TODO: This should be handled by a more generalized synchronization step.
-            var contextNames = GraphHandler
-                .GetNodes()
-                .Where(nodeHandler => nodeHandler.GetRegistryKey().Name == Registry.ResolveKey<ContextBuilder>().Name)
-                .Select(nodeHandler => nodeHandler.ID.LocalPath)
-                .ToList();
-
-            foreach (var localPath in contextNames)
-            {
-                try
-                {
-                    GraphHandler.ReconcretizeNode(localPath);
-                }
-                catch (Exception e)
-                {
-                    Debug.LogException(e);
-                }
-
-                if (!NodeModels.Any(nodeModel =>
-                        nodeModel is GraphDataContextNodeModel contextNodeModel &&
-                        contextNodeModel.graphDataName == localPath))
-                {
-                    this.CreateGraphDataContextNode(localPath);
-                }
-            }
+            this.CreateGraphDataContextNode("VertOut");
+            this.CreateGraphDataContextNode(ShaderGraphAssetUtils.kMainEntryContextNode);
```

Comment:
TODO:
SGM needs `SetActiveTarget(...)`: Calls `ConfigureWithTarget(...)` and then syncs `GraphDataContextNode`s accordingly.

GH needs `ConfigureWithTarget(...)`: Calls `AddContextNode(...)` and `RebuildContextData(...)` to match target requirements. This gets _really_ weird with cross pipeline, so we can keep this fake for URP.

GH needs a `GetEntryPoint()`: which gets the last context node in the chain.